### ISSUE TITLE
Very Slow e.g. Soft Delete via HasQueryFilter (Global OnModelCreating)

### ISSUE THREAD

**Issue by cgountanis** [dotnet/efcore#15996]
I enabled a HasQueryFilter on all tables that have the soft delete field which in my case is a DateTime?, only include IS NULL basically.

`modelBuilder.Entity<Account>().HasQueryFilter(p => p.DeletedDate == null);`

Problem is by doing this I am hitting upwards of 14,000ms on dbexecute on commands that did take 4ms. Obviously some of my context actions are including, thenincluding with mutiple children including reference tables. Is SQL really this slow at doing the sub-select thing?

What would be nice: The ability to do .Where on child objects 1>Many and still have it execute as a single query.

This is a pretty common thing in a bigger super relational/transactional database right? I do not want to get all the children in separate calls/queries if possible.

**Comment by ajcvickers** [comment]
@cgountanis We looked at this in triage but realized that we need a full repro. Can you post a small, runnable project/solution or complete code listing that demonstrates the behavior you are seeing.

**Comment by cgountanis** [comment]
I could probably whip up maybe a console app and a code first database that iterates some of the seeded test data. Possibly log to a rolling log file which should give the milliseconds per DB execute.

Maybe I could just start with one question. Will there ever be the ability to do a dot Where statement on an include/theninclude that contains a one to many relationship and have it run a single query?

**Comment by cgountanis** [comment]
I could probably whip up maybe a console app and a code first database that iterates some of the seeded test data. Possibly log to a rolling log file which should give the milliseconds per DB execute.

Maybe I could just start with one question. Will there ever be the ability to do a dot Where statement on an include/theninclude that contains a one to many relationship and have it run a single query?

**Comment by divega** [comment]
@cgountanis Include() with inline filters is something many customers have asked for but we haven't implemented yet. It is tracked in https://github.com/aspnet/EntityFrameworkCore/issues/1833. 

That said, we would like to get to the bottom of the performance issue you are seeing, because, as you guessed, query filters should provide the same functionality in some cases, and there is no well-known reason it should be slow.

**Comment by cgountanis** [comment]
Get back to you.

**Comment by cgountanis** [comment]
Working on a demo with data .BAK not a GIT, any way I can FTP or email when done, should be a fully repro project and data.

**Comment by cgountanis** [comment]
Working on a demo, will follow-up ASAP.

**Comment by ajcvickers** [comment]
**EF Team Triage:** Closing this issue as the requested additional details have not been provided and we have been unable to reproduce it.

*BTW this is a canned response and may have info or details that do not directly apply to this particular issue. While we'd like to spend the time to uniquely address every incoming issue, we get a lot traffic on the EF projects and that is not practical. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we use canned responses for common triage decisions.*

**Comment by cgountanis** [comment]
To be honest, just have not had the time to scope a demo with the relational DB/sample data needed. I tested in a console app on existing and did not have any issues, not sure why the MVC app would respond differently. Since then, we really just found a better way to break apart the EF calls into more modular repository calls vs. one big call with "things/joins/includes" we did not need. Looking forward to 3.0 as well, great work on the rewrite so far!
